### PR TITLE
feat: add animated button for cd modal

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
add animated button for cd modal

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/AnimatedDeployButton/AnimatedDeployButton.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                : {}
+                        }
+                    >
+                        <ICDeploy className="icon-dim-16" />
+                    </motion.div>
+                }
+                size={ComponentSizeType.large}
+                onClick={handleButtonClick}
+            />
+            {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
```
Comment (AbhishekA1509) [inline src/Shared/Components/AnimatedDeployButton/AnimatedDeployButton.tsx:66]: Check if aria-label will help

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -100,6 +100,7 @@ export interface customEnv {
     FEATURE_DEFAULT_MERGE_STRATEGY?: OverrideMergeStrategyType
     FEATURE_DEFAULT_LANDING_RB_ENABLE?: boolean
     FEATURE_CLUSTER_MAP_ENABLE?: boolean
+    FEATURE_ACTION_AUDIOS_ENABLE?: boolean
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/index.ts:103]: this is generic flag for audios throughout dashboard

File: src/Shared/Components/AnimatedDeployButton/AnimatedDeployButton.tsx [hunk 3]
```diff
@@ -16,25 +18,20 @@ const AnimatedDeployButton = ({ onButtonClick }: { onButtonClick: (e, disableDep
...
-    useEffect(
-        () => () => {
-            if (audioRef.current) {
-                audioRef.current.pause()
-            }
-        },
-        [],
-    )
+    useEffect(() => {
+        clearTimeout(timeoutRef.current)
```
Comment (AbhishekA1509) [inline src/Shared/Components/AnimatedDeployButton/AnimatedDeployButton.tsx:33]: I believe this must be unintentional

File: src/Shared/Components/AnimatedDeployButton/AnimatedDeployButton.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const [clicked, setClicked] = useState<boolean>(false)
+    const svgMotionVariants = {
+        hover: {
+            rotate: 45,
+        },
+    }
+
+    const handleButtonClick = async (e: SyntheticEvent) => {
+        if (isAudioEnabled && audioRef.current) {
+            await audioRef.current.play()
```
Comment (AbhishekA1509) [inline src/Shared/Components/AnimatedDeployButton/AnimatedDeployButton.tsx:23]: If async do we need to handleError

**Review by AbhishekA1509 — DISMISSED** [review]

**Review by AbhishekA1509 — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**